test_conformance/geometrics/test_geometrics_cross.cpp

```diff
@@ -0,0 +1,377 @@
+//
+// Copyright (c) 2022 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "test_geometrics_base.h"
+
+#include "harness/testHarness.h"
+#include "harness/typeWrappers.h"
+#include "harness/conversions.h"
+#include "harness/errorHelpers.h"
+#include <float.h>
+#include <limits>
+#include <cmath>
+
+#include <CL/cl_half.h>
+
+//--------------------------------------------------------------------------/
+// clang-format off
+// for readability sake keep this section unformatted
+
+static const char *crossKernelSource[] = {
+    GeomTestBase::extension,
+    "__kernel void sample_test(__global ", GeomTestBase::ftype, " *srcA, __global ", GeomTestBase::ftype, " *srcB, __global ", GeomTestBase::ftype, " *dst)\n"
+    "{\n"
+    "    int  tid = get_global_id(0);\n"
+    "    vstore%d( cross( vload%d( tid, srcA), vload%d( tid,  srcB) ), tid, dst );\n"
+    "\n"
+    "}\n"
+};
+
+// clang-format on
+//--------------------------------------------------------------------------
+
+struct CrossFPTest : public GeometricsFPTest
+{
+    CrossFPTest(cl_device_id d, cl_context c, cl_command_queue q)
+        : GeometricsFPTest(d, c, q)
+    {}
+    cl_int SetUp(int elements) override;
+    cl_int RunSingleTest(const GeomTestBase *p) override;
+
+    template <typename T> cl_int CrossKernel(const GeomTestParams<T> &p);
+};
+
+//--------------------------------------------------------------------------
+template <typename T>
```

Comment:
Would like the reference and error tolerance calculations in the two functions below to use the same type as the input, half for half and float for float as was the case earlier. 

Similarly for the reference and error tolerance calculation for dot product.